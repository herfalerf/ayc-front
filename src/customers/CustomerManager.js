import React, { useState, useEffect } from "react";
import { Link } from "react-router-dom";
import { Button } from "@material-ui/core";
import AycApi from "../api/api";
import CustomerCard from "./CustomerCard";
import LoadingSpinner from "../common/LoadingSpinner";

// Shows page with a list of customers with buttons to add/edit/delete custoemrs
//
// On Mount, loads customers from API.
//
// This is routed at /admin/customers
//
// Routes -> { Customers }
//

const CustomerManager = () => {
  console.debug("CustomerManager");

  const [customers, setCustomers] = useState(null);

  useEffect(() => {
    async function getCustomersOnMount() {
      console.debug("CustomerManager useEffect getCustomersOnMount");
      let customers = await AycApi.getCustomers();
      setCustomers(customers);
    }
    getCustomersOnMount();
  }, []);
  if (!customers) return <LoadingSpinner />;
  console.debug(customers);

  return (
    <div>
      <h1>Customer Manager</h1>
      <p>Here you can add, edit, or delete customers from the database</p>
      <Button
        variant="contained"
        component={Link}
        color="primary"
        to="/admin/customers/add"
      >
        Add A New Customer
      </Button>
      <div>
        {customers.map((c) => (
          <CustomerCard
            id={c.id}
            key={c.key}
            name={c.name}
            email={c.email}
            phone={c.phone}
            company={c.company}
          />
        ))}
      </div>
    </div>
  );
};

export default CustomerManager;
