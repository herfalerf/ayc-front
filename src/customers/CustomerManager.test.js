import React from "react";
import { render } from "@testing-library/react";
import CustomerManager from "./CustomerManager";
import { MemoryRouter } from "react-router";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <CustomerManager />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <CustomerManager />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
