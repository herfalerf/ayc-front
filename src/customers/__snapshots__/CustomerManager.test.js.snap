// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="makeStyles-upper-7"
    />
    <div
      class="MuiGrid-root makeStyles-customerHeader-8 MuiGrid-container MuiGrid-justify-content-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter"
        >
          Customer Manager
        </h4>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignCenter"
        >
          Here you can add, edit, or delete customers from the database
        </h6>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
        >
           
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            href="/admin/customers/add"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              Add A New Customer
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </p>
      </div>
    </div>
    <div
      class="LoadingSpinner"
    >
      Loading ...
    </div>
  </div>
</DocumentFragment>
`;
